libtransmission/session.cc

```diff
@@ -697,6 +699,15 @@ void tr_session::setSettings(tr_session_settings&& settings_in, bool force)
         tr_sessionSetCacheLimit_MB(this, val);
     }
 
+    if (auto const& val = new_settings.bind_address_ipv4; force || val != old_settings.bind_address_ipv4)
+    {
+        global_ip_cache_->set_settings_bind_addr(TR_AF_INET, new_settings.bind_address_ipv4);
+    }
+    if (auto const& val = new_settings.bind_address_ipv6; force || val != old_settings.bind_address_ipv6)
+    {
+        global_ip_cache_->set_settings_bind_addr(TR_AF_INET6, new_settings.bind_address_ipv6);
+    }
```

Comment:
There's another `bind_address_ipv4` / `bind_address_ipv6` block a couple of paragraphs further down (around line 735) and it looks like these two calls could be merged into that pre-existing code? Unless there's something in [710..735] that needs these lines to be called first?

Reply:
The `bind_address_ipv4` / `bind_address_ipv6` block below is enclosed within an `if (new_settings.tcp_enabled)`. We want `tr_global_ip_cache::set_settings_bind_addr()` to be called regardless of `tcp_enabled`.